pkg/controller/service/servicecontroller.go

```diff
@@ -787,3 +725,79 @@ func (s *cachedService) nextRetryDelay() time.Duration {
 func (s *cachedService) resetRetryDelay() {
 	s.lastRetryDelay = time.Duration(0)
 }
+
+// syncService will sync the Service with the given key if it has had its expectations fulfilled,
+// meaning it did not expect to see any more of its pods created or deleted. This function is not meant to be
+// invoked concurrently with the same key.
+func (s *ServiceController) syncService(key string) error {
+	startTime := time.Now()
+	var cachedService *cachedService
+	var retryDelay time.Duration
```

Comment:
The original logic is double the retry delay every time, so the delay time to put the element to queue is 1s, 2s, 4s, 8s. Here I kept the logic, we need some mechanism to calculate the delay time.